From 9489603484b4e44e4f4b30e70de463ab2faa88fa Mon Sep 17 00:00:00 2001
Message-ID: <9489603484b4e44e4f4b30e70de463ab2faa88fa.1737500778.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1737500778.git.sharlatanus@gmail.com>
References: <cover.1737500778.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 20 Jan 2025 21:31:01 +0000
Subject: [PATCH 13/27] gnu: python-numexpr: Update to 2.9.0.

* gnu/packages/python-xyz.scm (python-numexpr): Update to 2.9.0.
[native-inputs]: Add python-pytest, python-setuptools, and python-wheel.

Change-Id: I352536175d71e29c55c9ad71a96dbfd2b9b66759
---
 gnu/packages/python-xyz.scm | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/gnu/packages/python-xyz.scm b/gnu/packages/python-xyz.scm
index 41735a78ead..14ceb3b0fa5 100644
--- a/gnu/packages/python-xyz.scm
+++ b/gnu/packages/python-xyz.scm
@@ -9955,16 +9955,25 @@ (define-public python-numpydoc
 (define-public python-numexpr
   (package
     (name "python-numexpr")
-    (version "2.8.4")
+    (version "2.9.0") ; starting from 2.10.0, NumPy 2+ is required
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "numexpr" version))
        (sha256
-        (base32
-         "0iv1h1lvry5vmzqyvwxfbckyhzm1vbb1bmhmj4dnj64d84vjahym"))))
-    (build-system python-build-system)
-    (arguments `(#:tests? #f))          ; no tests included
+        (base32 "1w5ampdamlwj8ix1ipzxngmrlqpnmcmk95gbi6839kijqkv147gj"))))
+    (build-system pyproject-build-system)
+    (arguments
+     (list
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-before 'check 'build-extensions
+            (lambda _
+              (invoke "python" "setup.py" "build_ext" "--inplace"))))))
+    (native-inputs
+     (list python-pytest
+           python-setuptools
+           python-wheel))
     (propagated-inputs
      (list python-numpy))
     (home-page "https://github.com/pydata/numexpr")
-- 
2.47.1

